mautic/mautic
# Email sender by owner & email signature

### Background
Imagine a company with a sales department. Every salesman cares for a group of leads. In some cases (hopefully as a campaign action in the future) the lead can get another owner/salesman, but the lead still gets emails sent from the main email address, main from name and with default hard-coded signature in the emails.

### This PR adds 2 features:

#### 1. Sender email and name can change based on a lead owner
A new option in the Email Configuration can turn on this feature (turned off by default for BC). If a lead has some owner, then the lead gets emails from email address and from name from that owner/user.

#### 2. New `{signature}` token
There is a new option in the Email Configuration to define a default signature. Also at every user profile there is similar option. In an email can be used the `{signature}` token which will be replaced before send by the default signature or by the user signature if the lead owner is known.

### Testing
1. Apply this PR
2. Clear the cache (run `php app/console cache:clear --env dev`)
3. Update DB schema. (run `php app/console doctrine:schema:update`)
4. Make a list of for example 3 leads.
5. Create a new user if you have only one.
6. Set one of these leads as your lead, set second user as the owner of the second lead and leave the third lead without an owner to test all possible cases.
7. Go to configuration and turn the *Mailer is owner* switch to Yes.
8. Go to one of your leads and change the default signature to something else.
9. Create a new list email with for example this content:
```
<html>
<head>
	<title></title>
</head>
<body>
<p>Hello <strong contenteditable="false" data-token="{leadfield=firstname}">**Jm&eacute;no**</strong>!</p>

<p>Important message</p>

<p><strong contenteditable="false" data-token="{signature}">**Signature**</strong></p>

<p>the end</p>
</body>
</html>
```
Send the email and check that the email from, from name and the signature correspond to what you've configured.

Documentation for this PR: https://github.com/mautic/documentation/pull/29
